src/flb_output_thread.c

```diff
@@ -509,7 +509,7 @@ int flb_output_thread_pool_coros_size(struct flb_output_instance *ins)
 void flb_output_thread_pool_destroy(struct flb_output_instance *ins)
 {
     int n;
-    uint64_t stop = 0xdeadbeef;
+    struct flb_task *stop = 0xdeadbeef;
```

Comment:
`sizeof(stop)` used to evaluate to 8 bytes on 32 bit machines, however sizeof(struct flb_task *) would evaluate to 4 bytes on 32 bit machines. Thus, at the read end of the pipe, 0xdeadbeef, would be processed as 2, 4 byte tasks -- 1. 0xdeadbeef, and 2. 0x00000000. The latter creates a segfault when the worker thread processes this task.

This occurs when engine shutdown adds a 0xdeadbeef task indicating worker shutdown.

On 64 bit machines this is not a problem because (struct flb_task *) is 8 bytes, and so all 8 bytes are consumed as one task by the worker loop.